packages/@uppy/provider-views/src/ProviderView/ProviderView.js

```diff
@@ -430,11 +447,23 @@ module.exports = class ProviderView {
     }
   }
 
-  listAllFiles (path, files = null) {
+  listAllFiles (path, files = null, meta = undefined) {
     files = files || []
     return new Promise((resolve, reject) => {
       this.provider.list(path).then((res) => {
+        let itemsRelativePath
+
+        if (meta) {
```

Comment:
My idea on having this condition here is that if there's a meta, then there's a `previousRelativePath` and a `currentFolder`. That's why I skipped checking those individual properties.